### PR TITLE
add remote-function notebook fixes

### PR DESCRIPTION
**Pull request by svia3**
*Issue #, if available:*
-  HuggingFace error
-  Dev account id (https://github.com/aws/amazon-sagemaker-examples/blob/main/sagemaker-remote-function/pytorch_mnist_sample_script/config.yaml)
- Use random run name for Pytorch. UpdateTrialComponent fails the trainingJob if that run existed from a previous NB run.
- added workdir_config -> CustomFileFilter for the new for PyTorch example
```
ImportError: Using the `Trainer` with `PyTorch` requires `accelerate>=0.20.1`: Please run `pip install transformers[torch]` or `pip install accelerate -U`
```

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

- [x] I have read the [CONTRIBUTING](https://github.com/aws/amazon-sagemaker-examples/blob/master/CONTRIBUTING.md) doc and adhered to the example notebook best practices
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-sagemaker-examples/blob/master/README.md)
- [x] I have tested my notebook(s) and ensured it runs end-to-end
- [ ] I have linted my notebook(s) and code using `black-nb -l 100 {path}/{notebook-name}.ipynb`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: sagemaker-remote-function/pytorch_mnist_sample_script/config.yaml [hunk 1]
```diff
@@ -5,6 +5,6 @@ SageMaker:
       RemoteFunction:
         # role arn is not required if in SageMaker Notebook instance or SageMaker Studio
         # Uncomment the following line and replace with the right execution role if in a local IDE
-        RoleArn: arn:aws:iam::471094633457:role/SageMakerExecutionRole
+        # RoleArn: <replace the role arn here>
         InstanceType: ml.m5.xlarge
         Dependencies: ./requirements.txt
```
Comment (qidewenwhen) [inline sagemaker-remote-function/pytorch_mnist_sample_script/config.yaml:10]: How about this action item?
> Use workdir_config or the new for PyTorch example

Comment (svia3) [reply sagemaker-remote-function/pytorch_mnist_sample_script/config.yaml:10]: Ah good catch. But shouldn’t this be added to the new notebooks given it’s apart of the new lightsaber functionality ? + Do we have any current workdir_config examples that we can modify ?

Comment (navaj0) [reply sagemaker-remote-function/pytorch_mnist_sample_script/config.yaml:10]: It is for both @remote and @step.

Comment (navaj0) [inline sagemaker-remote-function/pytorch_mnist_sample_script/config.yaml:10]: Let's add 

```
IncludeLocalWorkDir: true
CustomFileFilter: 
    IgnoreNamePatterns:
        - "data"
    
```

File: sagemaker-remote-function/pytorch_mnist_sample_notebook/pytorch_mnist.ipynb [hunk 2]
```diff
@@ -572,11 +574,147 @@
...
-    "    experiment_name=\"pytorch-mnist-decorator\", run_name=\"run-1\", sagemaker_session=sm_session\n",
+    "    experiment_name=\"pytorch-mnist-decorator\", run_name=f\"run-{uuid.uuid4()}\", sagemaker_session=sm_session\n",
     ") as run:\n",
     "    trained_model = perform_train(train_set, test_set)"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "### Excluding current files/folder\n",
```
Comment (navaj0) [inline sagemaker-remote-function/pytorch_mnist_sample_notebook/pytorch_mnist.ipynb:585]: It is not a good place to show this capability here. 

    "        IncludeLocalWorkDir: true\n",
    "        CustomFileFilter:\n",

These two fields are mainly for users develop in a modular python package, and they need to package up the workspace and send to jobs. 

See my comments below.

File: sagemaker-remote-function/huggingface_text_classification/huggingface.ipynb [hunk 3]
```diff
@@ -172,7 +172,9 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "# train_dataset, test_dataset ="
+    "from datasets import load_dataset\n",
+    "\n",
+    "train_dataset, test_dataset = load_dataset('imdb', split=['train', 'test'])"
```
[the remark this answers is not in the document]
Comment (svia3) [reply sagemaker-remote-function/huggingface_text_classification/huggingface.ipynb:177]: Ah okay. I thought we had talked about this during one of our syncs, and Rohan had suggested we add it. I will remove it.

File: sagemaker-remote-function/pytorch_mnist_sample_script/config.yaml (hunk 1 above)
Comment (svia3) [reply sagemaker-remote-function/pytorch_mnist_sample_script/config.yaml:10]: Yea. So I assume if we add this example here as apart of this notebook, do we want to include something similar in our new notebooks for using `CustomFileFilter` with @steps

Comment (svia3) [reply sagemaker-remote-function/pytorch_mnist_sample_script/config.yaml:10]: Oh somehow the change to that file didn't make it to the git update. Weird, I had that.

File: sagemaker-remote-function/pytorch_mnist_sample_notebook/pytorch_mnist.ipynb [hunk 4]
```diff
@@ -362,8 +362,10 @@
    },
```
[the remark this answers is not in the document]
Comment (nmadan) [reply sagemaker-remote-function/pytorch_mnist_sample_notebook/pytorch_mnist.ipynb:362]: <p>intelligent defaults is an internal name. The external name is sagemaker config</p><p><a href="https://aws.amazon.com/blogs/machine-learning/configure-and-use-defaults-for-amazon-sagemaker-resources-with-the-sagemaker-python-sdk" target="_blank">https://aws.amazon.com/blogs/machine-learning/configure-and-use-defaults-for-amazon-sagemaker-resources-with-the-sagemaker-python-sdk/</a></p><p> </p>

[the remark this answers is not in the document]
Comment (svia3) [reply sagemaker-remote-function/pytorch_mnist_sample_notebook/pytorch_mnist.ipynb:362]: <p>Ah cool thanks. Will fix.</p>

**Pull request closed without merging**